powermon/device.py

```diff
@@ -96,21 +99,31 @@ def runLoop(self, force=False) -> bool:
         runs them
         """
         time.sleep(0.1)
-        if self.commands is None:
+        if self.commands is None or len(self.commands) == 0:
             log.info("no commands in queue")
             return False
         else:
             for command in self.commands:
                 if force or command.dueToRun():
-                    log.debug(f"Running command: {command.code}")
-                    # run command
-                    result: Result = self.port.run_command(command)
-                    # decode result
-                    self.port.get_protocol().decode(result=result, command=command)
-                    result.set_device_id(self.device_id)
-                    # loop through each output and process result
-                    output: AbstractOutput
-                    for output in command.outputs:
-                        log.debug(f"Using Output: {output}")
-                        output.process(result=result)
+                    #check if port need to be connected
+                    if not self.port.isConnected():
```

Comment:
add this check in the port, ie just do self.port.connect() here and then in the port itself check if anything needs to be done